sale_order_general_discount/models/sale_order_line.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    discount = fields.Float(
+        compute="_compute_general_discount",
+        store=True,
+        readonly=False,
+        string="Discount (%)",
+        digits="Discount",
+        default=0.0,
+    )
+
+    @api.depends("order_id")
+    def _compute_general_discount(self):
+        for line in self:
```

Comment:
For being friendly with inheritance, you must put:
```suggestion
        it hasattr(super(), "_compute_discount"):
            super()._compute_discount()
        for line in self:
```